docs/installation.rst

```diff
@@ -20,6 +20,38 @@ There are different ways to install |project_name|:
     have the latest code or use revision control (each release is
     tagged).
 
+.. _installation-requirements:
+
+Pre-Installation Considerations
+-------------------------------
+
+|project_name| Repository Filesystem
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- No special filename requirements exist. Filenames may contain any valid
+  ascii character and be of any length. |project_name| stores data only 3
+  directory levels deep and uses short directory and filenames.
+- No special file metadata requirements exist regarding permissions or extended
+  file attributes. |project_name| only requires read/write permissions within
+  the repository filesystem.
+- Segment files generated by |project_name| are no larger than 5MB by default
+  (Ver. 1.0.x). This can be modified in :ref:`config-file`
+- There are no special requirements with regard to inode limits. Data is stored
+  within the segment files per the size specified in the config. Storing many
+  small files will not translate into having an equal or larger amount of files
+  within the repository. Deduplication helps as well. Most importantly, the
+  number of segments within a directory can be set in the :ref:`config-file`.
+- Mkdir based file locking is used. File system based locking is not required.
+- |project_name| does not create hardlinks, symlinks or other special files
+- Written data must be consistent. **A journaling file system is strongly
```

Comment:
It was mentioned about symlinks/hardlinks in the original ticket. I considered that it didn't seem very important. However, in my research of backup systems, it seems some people either want hard linking (rsync) or don't want hard linking (don't want to use rsync). Therefore I thought it relevant to point out that Borg does not use hardlinks.